src/backend/tests/test_time_analyzer.py

```diff
@@ -634,3 +635,37 @@ def test_marijuana_violation_eligible_with_prior_conviction():
     assert isinstance(marijuana_violation.charge_type, MarijuanaViolation)
     assert expunger_result[marijuana_violation.ambiguous_charge_id].status is EligibilityStatus.ELIGIBLE
     assert expunger_result[prior_conviction.ambiguous_charge_id].status is EligibilityStatus.INELIGIBLE
+juvenile_doe_info = ["Juvenile Doe",
+                     str((date.today() + relativedelta(years=-22)).year)]
+
+def test_mj_under_21_not_blocked_by_traffic_violation():
+    print(juvenile_doe_info)
+    mj_violation_date = Time.FIVE_YEARS_AGO
+    marijuana_violation = ChargeFactory.create(
+        name="Possession of Marijuana < 1 Ounce",
+        statute="475860",
+        date=mj_violation_date,
+        disposition=DispositionCreator.create(ruling="Convicted", date=mj_violation_date),
+    )
+
+    case_1 = CaseFactory.create(
+        info=juvenile_doe_info,
+        type_status=["Juvenile Delinquency: Misdemeanor", "Closed"],
+        case_number="1",
+        charges=[marijuana_violation]
+    )
```

Comment:
I would just make a test with one Case and two charges. You may have to include the case number in the two charges like follows:
```
ChargeFactory.create(
        case_number="1",
```